### ISSUE TITLE
Pretraining not working on Apple Silicon, i.e. when device="mps"

### ISSUE THREAD

**Issue by shreyas269** [shreyas269/simple_llm#1]
The training loss saturates at a point most likely due to precision issues.

Simple fix could be to use autocast and gradient scaling.
